Library/OcCpuLib/OcCpuLib.c

```diff
@@ -273,33 +273,81 @@ ScanThreadCount (
 
 STATIC
 VOID
-ScanIntelFSBFrequency (
-  IN  OC_CPU_INFO        *CpuInfo
+SetMaxBusRatioAndMaxBusRatioDiv (
+  IN   OC_CPU_INFO        *CpuInfo  OPTIONAL,
+  OUT  UINT8              *MaxBusRatio,
+  OUT  UINT8              *MaxBusRatioDiv
   )
 {
   MSR_IA32_PERF_STATUS_REGISTER       PerfStatus;
   MSR_NEHALEM_PLATFORM_INFO_REGISTER  PlatformInfo;
-  UINT8                               MaxBusRatio;
-  UINT8                               MaxBusRatioDiv;
+  CPUID_VERSION_INFO_EAX              Eax;
+  UINT8                               CpuModel;
 
-  //
-  // TODO: this may not be accurate on some older processors.
-  //
-  if (CpuInfo->CpuGeneration >= OcCpuGenerationNehalem) {
-    PlatformInfo.Uint64 = AsmReadMsr64 (MSR_NEHALEM_PLATFORM_INFO);
-    MaxBusRatio = (UINT8) PlatformInfo.Bits.MaximumNonTurboRatio;
-    MaxBusRatioDiv = 0;
+  if (MaxBusRatio == NULL || MaxBusRatioDiv == NULL) {
```

Comment:
It is a requirement for this function to get pointers to `MaxBusRatio` and `MaxBusRatioDiv` (they are OUT, and there is no OPTIONAL). Thus these checks should become `ASSERT`s, because it is wrong to call the function with these pointers `NULL`.